cloudmarker/stores/mongodbstore.py

```diff
@@ -3,51 +3,67 @@
 import logging
 
-from pymongo import MongoClient
+from pymongo import MongoClient, errors
 
 _log = logging.getLogger(__name__)
 
 
 class MongoDBStore:
     """A plugin to store records on MongoDB."""
 
-    def __init__(self, host, port, username=None, password=None, db='gcp',
-                 buffer_size=1000):
+    def __init__(self, db, username, password, host, port=27017,
+                 buffer_size=1000, **kwargs):
         """Create an instance of :class:`MongoDBStore` plugin.
 
-        Arguments:
-            host (str): MongoDB instance host.
-            port (int): MongoDB instance port.
-            username (str): MongoDB instance auth username.
-            password (str): MongoDB instance auth password.
-            db (str): MongoDB database name, default: ``gcp``
-            buffer_size (int): Max number of records to hold in memeory for
-                each record_type.
+        It will use the default port for mongodb 27017 if not specified.
+        The Authentication scheme will be negotiated by MongoDB and the client
+        for v4.0+ to SCRAM-SHA-1 or SCRAM-SHA-256 by default aftere
+        negotiation.
 
+        Arguments:
+            db (str): name of the database
+            username (str): username for the database
+            password (str): password for username to authenticate with the db
+            host (str): hostname for the DB server
+            port (int): port for mongoDB is listening, defaults to 27017
+            buffer_size (int): max buffer before flushing to db
+            kwargs (dict): Additional args
+                * models - List of classes for validator and enforcement
+                    requirements.
         """
-        # pylint: disable=too-many-instance-attributes
-
-        self._mongodb_host = host
-        self._mongodb_port = port
-        self._mongodb_username = username
-        self._mongodb_password = password
-        self._db = db
+        self._client = MongoClient(
+            host=host,
+            port=port,
+            username=username,
+            password=password
+        )
+
+        self._db = self._client[db]
+
+        # Create collections if they don't exist based on the document
+        # restrictions the users want to have
+        """The collection creation, where the models objects specified, are
+        done with the following way. The object specified must have the
+        following functions:
+            collection(): returns the name of the collection
+            validator(): returns a valid string with $jsonSchema that contains
+                all the validation properties as specified in the mongodb
+                docs.
+            enforce(): returns one of the strings ``error`` or ``warn``.
+                This will be used to define the error level for when the schema
+                validation fails. ``error`` will raise an error while ``warn``
+                will log that violation in the db server logs.
+        """
```

Comment:
If it turns out that this change should indeed be merged, then the above comment needs to be reformatted with regular `#`-prefixed comments.

We reserve the usage of `"""`-quoted comments only for docstrings.